stl/inc/xutility

```diff
@@ -541,7 +541,8 @@ struct less_equal<void> {
 template <class _Fx>
 struct _Ref_fn { // pass function object by value as a reference
     template <class... _Args>
-    constexpr decltype(auto) operator()(_Args&&... _Vals) { // forward function call operator
+    constexpr decltype(auto) operator()(_Args&&... _Vals) noexcept(
+        _Select_invoke_traits<_Fx&, _Args...>::_Is_nothrow_invocable::value) { // forward function call operator

```

Comment:
No change required - I don't have data to back up the claim - but there's a lot of expensive SFINAE in `_Select_invoke_traits` that `_Ref_fn` doesn't need. We may be better off with:
```c++
template <class _Fx, bool = is_member_pointer_v<_Fx>>
struct _Ref_fn { // pass function object by value as a reference
    template <class... _Args>
    constexpr decltype(auto) operator()(_Args&&... _Vals) noexcept(
        noexcept(_STD invoke(_Fn, _STD forward<_Args>(_Vals)...))) {
        return _STD invoke(_Fn, _STD forward<_Args>(_Vals)...);
    }

    _Fx& _Fn;
};

template <class _Fx>
struct _Ref_fn<_Fx, false> {
    template <class... _Args>
    constexpr decltype(auto) operator()(_Args&&... _Vals) noexcept(noexcept(_Fn(_STD forward<_Args>(_Vals)...))) {
        return _Fn(_STD forward<_Args>(_Vals)...);
    }

    _Fx& _Fn;
};
```